infinispan-embedded/src/test/java/org/hibernate/ogm/datastore/infinispan/test/storedprocedures/InfinispanNamedParametersStoredProcedureCallTest.java

```diff
@@ -0,0 +1,119 @@
+
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.ParameterMode;
+import javax.persistence.PersistenceException;
+import javax.persistence.StoredProcedureQuery;
+
+import org.hibernate.ogm.backendtck.storedprocedures.Car;
+import org.hibernate.ogm.utils.TestForIssue;
+import org.hibernate.ogm.utils.jpa.OgmJpaTestCase;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.fest.assertions.Assertions.assertThat;
+import static org.hibernate.ogm.backendtck.storedprocedures.Car.RESULT_SET_PROC_ID_PARAM;
+import static org.hibernate.ogm.backendtck.storedprocedures.Car.RESULT_SET_PROC_TITLE_PARAM;
+
+/**
+ * Testing call of stored procedures
+ *
+ * @author The Viet Nguyen &amp;ntviet18@gmail.com&amp;
+ */
+@TestForIssue(jiraKey = { "OGM-1429" })
+public class InfinispanNamedParametersStoredProcedureCallTest extends OgmJpaTestCase {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	protected EntityManager em;
+
+	@Before
+	public void setUp() {
+		em = getFactory().createEntityManager();
+	}
+
+	@After
+	public void tearDown() {
+		em.close();
+	}
+
+	@Test
+	public void testResultSetDynamicCallWithFullQualifiedProcedureNameAndResultClass() throws Exception {
+		StoredProcedureQuery storedProcedureQuery = em.createStoredProcedureQuery( "org.hibernate.ogm.datastore.infinispan.test.storedprocedures.ResultSetProcedure", Car.class );
+		storedProcedureQuery.registerStoredProcedureParameter( "result", Void.class, ParameterMode.REF_CURSOR );
+		storedProcedureQuery.registerStoredProcedureParameter( RESULT_SET_PROC_ID_PARAM, Integer.class, ParameterMode.IN );
+		storedProcedureQuery.registerStoredProcedureParameter( RESULT_SET_PROC_TITLE_PARAM, String.class, ParameterMode.IN );
+		storedProcedureQuery.setParameter( RESULT_SET_PROC_ID_PARAM, 1 );
+		storedProcedureQuery.setParameter( RESULT_SET_PROC_TITLE_PARAM, "title" );
+		@SuppressWarnings("unchecked")
+		List<Car> listResult = storedProcedureQuery.getResultList();
+		assertThat( listResult ).containsExactly( new Car( 1, "title" ) );
+	}
+
+	@Test
+	public void testExceptionWhenUsePositionalParameters() throws Exception {
+		thrown.expect( PersistenceException.class );
+		thrown.expectMessage( "org.hibernate.HibernateException: OGM001115: Dialect org.hibernate.ogm.datastore.infinispan.impl.InfinispanEmbeddedStoredProceduresDialect does not support positional parameters when calling stored procedures" );
+		StoredProcedureQuery storedProcedureQuery = em.createStoredProcedureQuery( Car.SIMPLE_VALUE_PROC, Car.class );
+		storedProcedureQuery.registerStoredProcedureParameter( 0, Integer.class, ParameterMode.IN );
+		storedProcedureQuery.setParameter( 0, 1 );
+		storedProcedureQuery.getSingleResult();
+	}
+
+	@Test
+	public void testExceptionWhenInvalidProcedureName() throws Exception {
```

Comment:
Invalid is a very vague statement, is it invalid because it doesn't exists on the server?

Reply:
In that case we should explain the we haven't found the store procedure on the server or something like that